pkg/cmd/oauth-apiserver/cmd.go

```diff
@@ -111,5 +116,46 @@ func (o *OAuthAPIServerOptions) NewOAuthAPIServer() (*apiserver.OAuthAPIServer,
 		return nil, err
 	}
 
+	etcdOptions := o.RecommendedOptions.Etcd
+	etcdOptions.DefaultStorageMediaType = "application/vnd.kubernetes.protobuf"
+
+	storageFactory := apiserverstorage.NewDefaultStorageFactory(
+		etcdOptions.StorageConfig,
+		etcdOptions.DefaultStorageMediaType,
+		legacyscheme.Codecs,
+		apiserverstorage.NewDefaultResourceEncodingConfig(legacyscheme.Scheme),
+		&serverstorage.ResourceConfig{},
+		specialDefaultResourcePrefixes,
+	)
+
+	if len(etcdOptions.EncryptionProviderConfigFilepath) != 0 {
+		transformerOverrides, err := encryptionconfig.GetTransformerOverrides(etcdOptions.EncryptionProviderConfigFilepath)
+		if err != nil {
+			return nil, err
+		}
+		for groupResource, transformer := range transformerOverrides {
+			storageFactory.SetTransformer(groupResource, transformer)
+		}
+	}
+	if err := o.RecommendedOptions.Etcd.ApplyWithStorageFactoryTo(storageFactory, &serverConfig.GenericConfig.Config); err != nil {
+		return nil, err
+	}
+
 	return serverConfig.Complete().New(genericapiserver.NewEmptyDelegate())
 }
+
+// specialDefaultResourcePrefixes are a custom storage prefixes (we must be backward compatible with OpenShift API)
+var specialDefaultResourcePrefixes = map[schema.GroupResource]string{
+	{Resource: "oauthaccesstokens"}:                                 "oauth/accesstokens",
+	{Resource: "oauthaccesstokens", Group: "oauth.openshift.io"}:    "oauth/accesstokens",
+	{Resource: "oauthauthorizetokens"}:                              "oauth/authorizetokens",
+	{Resource: "oauthauthorizetokens", Group: "oauth.openshift.io"}: "oauth/authorizetokens",
+
+	{Resource: "oauthclients"}:                                           "oauth/clients",
+	{Resource: "oauthclients", Group: "oauth.openshift.io"}:              "oauth/clients",
+	{Resource: "oauthclientauthorizations"}:                              "oauth/clientauthorizations",
+	{Resource: "oauthclientauthorizations", Group: "oauth.openshift.io"}: "oauth/clientauthorizations",
+
+	{Resource: "identities"}:                             "useridentities",
+	{Resource: "identities", Group: "user.openshift.io"}: "useridentities",
```

Comment:
do we access these?

Reply:
the new server will manage `user.openshift.io`

Reply:
oh, right.